minimatch.js

```diff
@@ -708,11 +698,6 @@ function match (f, partial) {
 
   var options = this.options
 
-  // windows: need to use /, not \
-  if (path.sep !== '/') {
```

Comment:
Pattern (as some kind of special language) SHOULD be cross-platform, and therefore need to use one special character (forward slash) as path separator on all platforms.
But, in opposite, file path is platform dependend by definition .
For example, code:
`if ( minimatch(filepath, 'foo/*.js') ) { /* do something*/` }
will perfectly works with current `minimatch` version (or with my fork) on all platforms.
But with your commit the condition will do never trigger on Windows.

Moreover, imagine that one day will be invented a new operation system, and it developers decide to use pipe characted ('|') as path separator. The code example will work on this OS without any changes, because filepath `foo|bar.js` will be internally replaced with `foo/bar.js` (using code that you try to remove) and successfully match to the pattern 'foo/*.js'